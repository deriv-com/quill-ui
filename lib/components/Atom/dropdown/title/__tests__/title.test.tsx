import React from "react";
import { render } from "@testing-library/react";
import DropdownTitle from "..";

describe("DropdownTitle", () => {
    it("renders with default div element", () => {
        const { container } = render(<DropdownTitle label="Test Label" />);
        expect(container.firstChild).toHaveClass(
            "quill-dropdown-title__container",
        );
        expect(container.firstChild?.nodeName).toBe("DIV");
    });

    it("renders with specified element", () => {
        const { container } = render(
            <DropdownTitle as="section" label="Test Label" />,
        );
        expect(container.firstChild).toHaveClass(
            "quill-dropdown-title__container",
        );
        expect(container.firstChild?.nodeName).toBe("SECTION");
    });

    it("applies size class", () => {
        const { container } = render(
            <DropdownTitle size="sm" label="Test Label" />,
        );
        expect(container).toMatchSnapshot();
    });

    it("applies size class", () => {
        const { container } = render(
            <DropdownTitle size="md" label="Test Label" />,
        );
        expect(container).toMatchSnapshot();
    });

    it("renders label", () => {
        const { getByText } = render(<DropdownTitle label="Test Label" />);
        expect(getByText("Test Label")).toBeInTheDocument();
    });

    it("renders icon", () => {
        const { getByText } = render(
            <DropdownTitle label="Test Label" icon={<span>Icon</span>} />,
        );
        expect(getByText("Icon")).toBeInTheDocument();
    });

    it("applies centered class", () => {
        const { container } = render(
            <DropdownTitle centered label="Test Label" />,
        );
        expect(container).toMatchSnapshot();
    });

    it("forwards rest props to the container element", () => {
        const { container } = render(
            <DropdownTitle label="Test Label" data-testid="dropdown-title" />,
        );
        expect(container.firstChild).toHaveAttribute("data-testid");
    });
});
