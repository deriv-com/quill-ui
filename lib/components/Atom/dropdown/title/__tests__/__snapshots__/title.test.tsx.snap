// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownTitle applies centered class 1`] = `
<div>
  <div
    class="quill-dropdown-title__container quill-dropdown-title__container--md"
  >
    <p
      class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--subtle quill-typography-centered quill-dropdown-title__label"
    >
      Test Label
    </p>
  </div>
</div>
`;

exports[`DropdownTitle applies size class 1`] = `
<div>
  <div
    class="quill-dropdown-title__container quill-dropdown-title__container--sm"
  >
    <p
      class="quill-typography quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--subtle quill-dropdown-title__label"
    >
      Test Label
    </p>
  </div>
</div>
`;

exports[`DropdownTitle applies size class 2`] = `
<div>
  <div
    class="quill-dropdown-title__container quill-dropdown-title__container--md"
  >
    <p
      class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--subtle quill-dropdown-title__label"
    >
      Test Label
    </p>
  </div>
</div>
`;
